datafusion/core/src/physical_plan/file_format/parquet/page_filter.rs

```diff
@@ -99,94 +100,110 @@ use super::metrics::ParquetFileMetrics;
 ///
 /// So we can entirely skip rows 0->199 and 250->299 as we know they
 /// can not contain rows that match the predicate.
-pub(crate) fn build_page_filter(
-    pruning_predicate: Option<&PruningPredicate>,
-    schema: SchemaRef,
-    row_groups: &[usize],
-    file_metadata: &ParquetMetaData,
-    file_metrics: &ParquetFileMetrics,
-) -> Result<Option<RowSelection>> {
-    // scoped timer updates on drop
-    let _timer_guard = file_metrics.page_index_eval_time.timer();
-    let page_index_predicates =
-        extract_page_index_push_down_predicates(pruning_predicate, schema)?;
+#[derive(Debug)]
+pub(crate) struct PagePruningPredicate {
+    predicates: Vec<PruningPredicate>,
+}
 
-    if page_index_predicates.is_empty() {
-        return Ok(None);
+impl PagePruningPredicate {
+    /// Create a new [`PagePruningPredicate`]
+    pub fn try_new(expr: &Expr, schema: SchemaRef) -> Result<Self> {
+        let predicates = split_conjunction(expr)
+            .into_iter()
+            .filter_map(|predicate| {
+                let columns = predicate.to_columns().ok()?;
+                (columns.len() == 1)
+                    .then(|| PruningPredicate::try_new(predicate.clone(), schema.clone()))
+            })
+            .collect::<Result<Vec<_>>>()?;
+        Ok(Self { predicates })
     }
 
-    let groups = file_metadata.row_groups();
+    /// Returns a [`RowSelection`] for the given file
+    pub fn prune(
```

Comment:
I recommend viewing this with whitespace disabled

https://github.com/apache/arrow-datafusion/pull/4713/files?diff=split&w=1